### ISSUE TITLE
Bug: eslint fails with multi-filetype config

### ISSUE THREAD

**Issue by John-L** [eslint/eslint#19819]
### Environment

Node version: v22.16.0
npm version: 10.9.2
Local ESLint version: 9.28.0
Global ESLint version: Not found
Operating System: win32 10.0.26100

### What parser are you using?

Default (Espree)

### What did you do?

<details>
`npx eslint Test.html "Test (2).js"`

<summary>Configuration</summary>

```eslint.config.mjs
import { defineConfig } from "eslint/config";
import path from "node:path";
import { fileURLToPath } from "node:url";
import js from "@eslint/js";
import html from "@html-eslint/eslint-plugin";
import { FlatCompat } from "@eslint/eslintrc";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const compat = new FlatCompat({
  baseDirectory: __dirname,
  recommendedConfig: js.configs.recommended,
  allConfig: js.configs.all,
});

export default defineConfig([
  {
    files: ["**/*.js", "**/*.html"],
    extends: compat.extends("prettier"),
  },
  // lint js files
  {
    files: ["**/*.js"],
    plugins: {
      js,
    },
    language: ["js/js"],
    extends: ["js/recommended"],
    rules: {
      "prefer-const": "error",
      "no-unused-vars": "warn",
    },
  },
  // lint html files
  {
    files: ["**/*.html"],
    plugins: {
      html,
    },
    language: ["html/html"],
    extends: ["html/recommended"],
  },
]);
```
</details>

```Test (2).js
testA = "22pt";
testB = "6pt";
window.alert((testA > testB) + " " + (testA < testB));
document.getElementById("testa").style.display = "inline";
document.getElementById("testa").style.backgroundColor = "indigo";
document.getElementById("testa").style.color = "white";
document.getElementById("testa").style.fontSize = testA;
document.getElementById("testa").style.display = "inline";
document.getElementById("testb").style.backgroundColor = "indigo";
document.getElementById("testb").style.color = "white";
document.getElementById("testa").style.fontSize = testA;
```

```Test.html
<html>
  <body></body>
  <span id="heading1">Hello </span>
  <span id="heading2">world</span>
  <span id="heading3">.</span>
  <script>
    document.getElementById("heading1").style.color = "blue";
    document.getElementById("heading2").style.color = "red";
    document.getElementById("heading3").style.color = "white";
  </script>
</html>
```

### What did you expect to happen?

eslint to run to completion

### What actually happened?

Oops! Something went wrong! :(

ESLint: 9.28.0

TypeError: identifier.split is not a function
    at splitPluginIdentifier (C:\Users\johnl\OneDrive\Projects\Project B\node_modules\eslint\lib\config\config.js:223:27)
    at new Config (C:\Users\johnl\OneDrive\Projects\Project B\node_modules\eslint\lib\config\config.js:405:7)
    at [finalizeConfig] (C:\Users\johnl\OneDrive\Projects\Project B\node_modules\eslint\lib\config\flat-config-array.js:212:10)
    at FlatConfigArray.getConfigWithStatus (C:\Users\johnl\OneDrive\Projects\Project B\node_modules\@eslint\config-array\dist\cjs\index.cjs:1263:55)
    at FlatConfigArray.getConfig (C:\Users\johnl\OneDrive\Projects\Project B\node_modules\@eslint\config-array\dist\cjs\index.cjs:1281:15)
    at C:\Users\johnl\OneDrive\Projects\Project B\node_modules\eslint\lib\eslint\eslint.js:762:28
    at async Promise.all (index 0)
    at async ESLint.lintFiles (C:\Users\johnl\OneDrive\Projects\Project B\node_modules\eslint\lib\eslint\eslint.js:758:19)
    at async Object.execute (C:\Users\johnl\OneDrive\Projects\Project B\node_modules\eslint\lib\cli.js:637:14)
    at async main (C:\Users\johnl\OneDrive\Projects\Project B\node_modules\eslint\bin\eslint.js:175:19)
npm notice
npm notice New major version of npm available! 10.9.2 -> 11.4.1
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.4.1
npm notice To update run: npm install -g npm@11.4.1
npm notice

### Link to Minimal Reproducible Example

https://github.com/John-L/Project-B

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

Unclear if the root source of the error is my config file contents, or eslint code.

**Comment by sethamus** [comment]
The `language` key has to be a string, not an array. Try this instead:

<details>
<summary>Configuration</summary>

```js
import { defineConfig } from "eslint/config";
import path from "node:path";
import { fileURLToPath } from "node:url";
import js from "@eslint/js";
import html from "@html-eslint/eslint-plugin";
import { FlatCompat } from "@eslint/eslintrc";

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);
const compat = new FlatCompat({
  baseDirectory: __dirname,
  recommendedConfig: js.configs.recommended,
  allConfig: js.configs.all,
});

export default defineConfig([
  {
    files: ["**/*.js", "**/*.html"],
    extends: compat.extends("prettier"),
  },
  // lint js files
  {
    files: ["**/*.js"],
    plugins: {
      js,
    },
    extends: ["js/recommended"],
    rules: {
      "prefer-const": "error",
      "no-unused-vars": "warn",
    },
  },
  // lint html files
  {
    files: ["**/*.html"],
    plugins: {
      html,
    },
    language: "html/html",
    extends: ["html/flat/recommended"],
  },
]);
```
</details>
